lancero_source.go

```diff
@@ -336,12 +336,16 @@ func (ls *LanceroSource) PrepareChannels() error {
 	ls.rowColCodes = make([]RowColCode, ls.nchan)
 	ls.chanNames = make([]string, ls.nchan)
 	ls.chanNumbers = make([]int, ls.nchan)
+	ls.subframeOffsets = make([]int, ls.nchan)
 	index := 0
 	cnum := ls.firstRowChanNum
 	thisColFirstCnum := cnum - ls.chanSepColumns
 	ls.groupKeysSorted = make([]GroupIndex, 0)
 	for _, device := range ls.active {
-		ls.rowFrameRatio = device.nrows
+		// For Lancero sources, subframeDivisions = the number of rows.
+		// For sources with multiple LanceroDevice objects, its meaning is ambiguous, but we'll
+		// take the max value of all numbers of rows and hope for the best.
```

Comment:
Assert that they all have the same number of rows, throw an error if not. Then use that number of rows.

Reply:
I cannot see ever exercising this case (would be reading 2+ lancero devices at once, with unequal # of rows). You convinced me that an assertion here is superior to what I was trying (a placeholder that would lead to great confusion if ever actually used). I'll assert/panic.

The **real** requirement here would be to treat external triggers as a per-device property, not a global property of the `LanceroSource`. But then we'd have to figure out how to write multiple external trigger files, which is a condition we _never_ anticipate using in the real world. Better to assert equality and panic if not present.